### PR TITLE
feat: make turrets and Runes use owner UUID

### PR DESCRIPTION
**Pull request by dphaldes**
Supersedes #1422 

Modifies ANFakePlayer to store player UUID. Turrets and Runes store the uuid and wrap it in their caster.

Also updated
- EffectPlaceBlock 
- EffectCut

### CODE REVIEW

File: src/main/java/com/hollingsworth/arsnouveau/common/items/SpellCrossbow.java [hunk 1]
```diff
@@ -81,14 +82,18 @@ public void releaseUsing(@NotNull ItemStack pStack, @NotNull Level pLevel, @NotN
     }
 
     private boolean tryLoadProjectiles(LivingEntity pShooter, ItemStack pCrossbowStack) {
+        if(pShooter.level().isClientSide){
```
Comment (baileyholl) [inline src/main/java/com/hollingsworth/arsnouveau/common/items/SpellCrossbow.java:85]: What is this for? Vanilla does not side check here for the bow, is this fixing some other behavior?

Comment (mystchonky) [reply src/main/java/com/hollingsworth/arsnouveau/common/items/SpellCrossbow.java:85]: if you look down a bit, you could see that I am trying to cast Level into ServerLevel. This method is called on both sides (I think for animation purposes and probably other things) so this check ensures that spell related code is run only on the server

File: src/main/java/com/hollingsworth/arsnouveau/common/spell/effect/EffectFell.java [hunk 2]
```diff
@@ -43,14 +43,14 @@ public void onResolveBlock(BlockHitResult ray, Level world, @NotNull LivingEntit
...
                 if (spellStats.hasBuff(AugmentExtract.INSTANCE)) {
                     world.getBlockState(listPos).getDrops(LootUtil.getSilkContext((ServerLevel) world, listPos, shooter)).forEach(i -> world.addFreshEntity(new ItemEntity(world, listPos.getX(), listPos.getY(), listPos.getZ(), i)));
-                    BlockUtil.destroyBlockSafelyWithoutSound(world, listPos, false);
+                    BlockUtil.destroyBlockSafelyWithoutSound(world, listPos, false, shooter);
                 } else if (spellStats.hasBuff(AugmentFortune.INSTANCE)) {
                     world.getBlockState(listPos)
                             .getDrops(LootUtil.getFortuneContext((ServerLevel) world, listPos, shooter, spellStats.getBuffCount(AugmentFortune.INSTANCE)))
                             .forEach(i -> world.addFreshEntity(new ItemEntity(world, listPos.getX(), listPos.getY(), listPos.getZ(), i)));
-                    BlockUtil.destroyBlockSafelyWithoutSound(world, listPos, false);
+                    BlockUtil.destroyBlockSafelyWithoutSound(world, listPos, false, shooter);
```
Comment (baileyholl) [inline src/main/java/com/hollingsworth/arsnouveau/common/spell/effect/EffectFell.java:51]: This is going to cause dupes as the above code is making drops without a promise the block was destroyed. We have to check that the block was successfully destroyed first.

File: src/main/java/com/hollingsworth/arsnouveau/common/ritual/RitualAwakening.java [hunk 3]
```diff
@@ -19,26 +20,37 @@
...
+    @Override
+    public void onStart(@Nullable Player player) {
+        if (player != null) uuid = player.getUUID();
+    }
+
+    public void destroyTree(ServerLevel world, Set<BlockPos> set) {
+        FakePlayer fakePlayer = ANFakePlayer.getPlayer(world, uuid);
         for (BlockPos p : set) {
-            BlockUtil.destroyBlockSafelyWithoutSound(world, p, false);
+            BlockUtil.destroyBlockSafelyWithoutSound(world, p, false, fakePlayer);
```
Comment (baileyholl) [inline src/main/java/com/hollingsworth/arsnouveau/common/ritual/RitualAwakening.java:49]: This is technically causing a dupe here if we aren't going to check that a block was successfully destroyed before we spawn the entity.

**Pull request merged**